server/routes/overview/viewGoalsController.ts

```diff
@@ -1,11 +1,29 @@
 import { RequestHandler } from 'express'
 import ViewGoalsView from './viewGoalsView'
+import EducationAndWorkPlanService from '../../services/educationAndWorkPlanService'
 
 export default class ViewGoalsController {
-  constructor() {}
+  constructor(private readonly educationAndWorkPlanService: EducationAndWorkPlanService) {}
 
-  viewGoals: RequestHandler = async (req, res, next): Promise<void> => {
-    const view = new ViewGoalsView(req.session.prisonerSummary)
-    res.render('pages/overview/partials/goalsTab/goalsTabContents', { ...view.renderArgs })
+  viewGoals: RequestHandler = async (req, res, next) => {
+    try {
+      const { prisonerSummary } = req.session
+      const { prisonNumber } = req.params
+      const { username } = req.user
+
+      const prisonerGoals = await this.educationAndWorkPlanService.getAllGoalsForPrisoner(prisonNumber, username)
```

Comment:
No need to call the service to get the prisoner goals, because you've already done it in the route definition via your new middleware.
At this point in your code you can safely assume that res.locals contains a `PrisonerGoals` instance (res.locals.allGoalsForPrisoner)

Reply:
I also dont think we'll need the try catch 👍